With covid-19 beign categorized as the pandemic of the decade it has become necessary for the community ranging from scientist to cloud providers to government of the countries for the urgent and valid analysis of the virus spread.<br><br>
The dataset which I used is available at kaggle. I have provided the analysis of the affect of the virus on the Indian states. Following were the main points which I have covered using numpy and pandas module for the analysis:-
1. Is there any missing value in data.
2. The size shape and dimension of the dataset.
3. The total sum of the different cases emerged.
4. Mean, median, percentile and standard division provided the EDA analysis of the dataset.
5. The states having Total Confirmed cases more than mean.
6. The states having death rate more than mean.
7. The states having Cured/Discharged/Migrated less than mean.
8. The maximum values of different cases.
9. The minimum values of different cases.
10. The ratio of Deaths to Total Confirmed cases in respective states provide the fatality ratio of the state.
11. The states having the fatality rate less than the mean.
12. The ratio of Cured to Total Confirmed cases in respective states provide the mortality ratio of the state.
13. The states having the mortality rate more than the mean.

In [1]:
import pandas as pd
import numpy as np

# Importing the data


In [2]:
file = pd.read_csv('CovidIndia.csv')

In [3]:
file.head()

S. No.           Name of State / UT  Total Confirmed cases  Active  \
0       1               Andhra Pradesh                   1583    1062   
1       2  Andaman and Nicobar Islands                     33       1   
2       3            Arunachal Pradesh                      1       0   
3       4                        Assam                     43       9   
4       5                        Bihar                    517     396   

   Cured/Discharged/Migrated  Deaths  
0                        488      33  
1                         32       0  
2                          1       0  
3                         33       1  
4                        117       4

# Checking the null value.

In [4]:
is_nan = file.isnull().values.any()
is_nan

False

# Checking the size, shape and dimension of the array

In [5]:
file_size = file.size
print('The size of array is: ', file_size, sep = '\n')
file_shape = file.shape
print('\nThe shape of array is: ',file_shape, sep = '\n')
file_dim = file.ndim
print('\nThe dimension of array is: ', file_dim, sep = '\n')

The size of array is: 
192

The shape of array is: 
(32, 6)

The dimension of array is: 
2


# The sum of different individual cases.

In [6]:
sum_analysis = np.sum(file, axis = 0)
print('''The sum of the columns ''', sum_analysis, sep = '\n')

The sum of the columns 
S. No.                                                                     528
Name of State / UT           Andhra PradeshAndaman and Nicobar IslandsAruna...
Total Confirmed cases                                                    42546
Active                                                                   29376
Cured/Discharged/Migrated                                                11775
Deaths                                                                    1391
dtype: object


# The mean of different cases

In [7]:
mean_analysis = np.mean(file, axis = 0)
print('The mean of the columns is: ', mean_analysis, sep = '\n')

The mean of the columns is: 
S. No.                         16.50000
Total Confirmed cases        1329.56250
Active                        918.00000
Cured/Discharged/Migrated     367.96875
Deaths                         43.46875
dtype: float64


# States where the Total Confirmed cases are more than mean cases

I tried various numpy method but I was not able to produce the below output. Hence I used pandas. Please let me know if you have the way to produce the following output using numpy.

In [8]:
cases_greater_than_mean = file.loc[file["Total Confirmed cases"] >= 1329.56250]
print('Following chart shows the state where cases are greater than mean: ')
cases_greater_than_mean

Following chart shows the state where cases are greater than mean: 


S. No. Name of State / UT  Total Confirmed cases  Active  \
0        1     Andhra Pradesh                   1583    1062   
7        8              Delhi                   4549    3123   
9       10            Gujarat                   5428    4096   
17      18     Madhya Pradesh                   2837    1883   
18      19        Maharashtra                  12974   10311   
25      26          Rajasthan                   2886    1459   
26      27         Tamil Nadu                   3023    1614   
30      31      Uttar Pradesh                   2645    1848   

    Cured/Discharged/Migrated  Deaths  
0                         488      33  
7                        1362      64  
9                        1042     290  
17                        798     156  
18                       2115     548  
25                       1356      71  
26                       1379      30  
30                        754      43

# States where the Death cases are more than mean 

In [9]:
deaths_greater_than_mean = file.loc[file["Deaths"] >= 43.46875]
print('Following chart shows the state where deaths are greater than mean: ')
deaths_greater_than_mean

Following chart shows the state where deaths are greater than mean: 


S. No. Name of State / UT  Total Confirmed cases  Active  \
7        8              Delhi                   4549    3123   
9       10            Gujarat                   5428    4096   
17      18     Madhya Pradesh                   2837    1883   
18      19        Maharashtra                  12974   10311   
25      26          Rajasthan                   2886    1459   
31      32        West Bengal                    963     762   

    Cured/Discharged/Migrated  Deaths  
7                        1362      64  
9                        1042     290  
17                        798     156  
18                       2115     548  
25                       1356      71  
31                        151      50

# States where the Cured/Discharged/Migrated cases are less than mean 

In [10]:
cured_less_than_mean = file.loc[file["Cured/Discharged/Migrated"] <= 367.96875]
print('Following chart shows the state where Cured/Discharged/Migrated are less than mean: ')
cured_less_than_mean.head()

Following chart shows the state where Cured/Discharged/Migrated are less than mean: 


S. No.           Name of State / UT  Total Confirmed cases  Active  \
1       2  Andaman and Nicobar Islands                     33       1   
2       3            Arunachal Pradesh                      1       0   
3       4                        Assam                     43       9   
4       5                        Bihar                    517     396   
5       6                   Chandigarh                     97      77   

   Cured/Discharged/Migrated  Deaths  
1                         32       0  
2                          1       0  
3                         33       1  
4                        117       4  
5                         19       1

# Calculating median along the axis

In [11]:
selecting_rows = file.iloc[:,2:].values
median_analysis = np.median(selecting_rows, axis = 0)
print('The median along the axis 0 is: ', median_analysis, sep = '\n')

The median along the axis 0 is: 
[302.   98.   88.5   3.5]


# Checking percentile 50, 75.

In [12]:
checking_percentile = np.percentile(selecting_rows, 50, axis = 0)
print('The percentile along the axis 0 is: ', checking_percentile, sep = '\n')

The percentile along the axis 0 is: 
[302.   98.   88.5   3.5]


As we can see above the value of percentile and median are same. This means that there are half of the active cases less than 302.

In [13]:
checking_percentile_75 = np.percentile(selecting_rows, 75, axis = 0)
print('The 75 percentile along the axis 0 is: ', checking_percentile_75, sep = '\n')

The 75 percentile along the axis 0 is: 
[1222.25  988.5   502.25   30.75]


The above values denote that the 25% values are less than above result.

# Checking standard deviation

In [14]:
file_std = np.std(selecting_rows, axis = 0)
print('The standard deviation are as follows', file_std, sep = '\n')

The standard deviation are as follows
[2511.77152297 1944.14119858  530.69968464  106.91643477]


# Checking the maximum values for each case

In [15]:
max_analysis = np.max(file, axis = 0)
print('Following shows the maximum cases: ', max_analysis, sep = '\n')

Following shows the maximum cases: 
S. No.                                32
Name of State / UT           West Bengal
Total Confirmed cases              12974
Active                             10311
Cured/Discharged/Migrated           2115
Deaths                               548
dtype: object


# Checking the minimum values for each case

In [16]:
min_analysis = np.min(file, axis = 0)
print('Following shows the minimum cases: ', min_analysis, sep = '\n')

Following shows the minimum cases: 
S. No.                                                 1
Name of State / UT           Andaman and Nicobar Islands
Total Confirmed cases                                  1
Active                                                 0
Cured/Discharged/Migrated                              0
Deaths                                                 0
dtype: object


# Determining the fatality ratio

In [17]:
fatality_ratio = 42546 / 1391
print('The death rate is: ', fatality_ratio, sep = '\n')

The death rate is: 
30.586628324946084


In [18]:
total_case = selecting_rows[:,0:-3]
death_rows = selecting_rows[:,-1:]
fatality_ratio_state =  death_rows / total_case 

In [19]:
fatality_1 = np.hstack([selecting_rows, fatality_ratio_state])

print('The new array with volume column is: ', fatality_1[:4], sep = '\n')

The new array with volume column is: 
[[1.5830000e+03 1.0620000e+03 4.8800000e+02 3.3000000e+01 2.0846494e-02]
 [3.3000000e+01 1.0000000e+00 3.2000000e+01 0.0000000e+00 0.0000000e+00]
 [1.0000000e+00 0.0000000e+00 1.0000000e+00 0.0000000e+00 0.0000000e+00]
 [4.3000000e+01 9.0000000e+00 3.3000000e+01 1.0000000e+00 2.3255814e-02]]


In [20]:
file['Fatality rate'] = fatality_ratio_state
file.head()

S. No.           Name of State / UT  Total Confirmed cases  Active  \
0       1               Andhra Pradesh                   1583    1062   
1       2  Andaman and Nicobar Islands                     33       1   
2       3            Arunachal Pradesh                      1       0   
3       4                        Assam                     43       9   
4       5                        Bihar                    517     396   

   Cured/Discharged/Migrated  Deaths  Fatality rate  
0                        488      33       0.020846  
1                         32       0       0.000000  
2                          1       0       0.000000  
3                         33       1       0.023256  
4                        117       4       0.007737

In [21]:
fatality_mean = np.mean(file, axis = 0)
print('The mean of fatality rate is: ', fatality_mean, sep = '\n')

The mean of fatality rate is: 
S. No.                         16.50000
Total Confirmed cases        1329.56250
Active                        918.00000
Cured/Discharged/Migrated     367.96875
Deaths                         43.46875
Fatality rate                   0.01966
dtype: float64


# Determing the states having less fatality ratio than average

In [22]:
fatality_more_than_mean = file.loc[file["Fatality rate"] < 0.01966]
print('The states with fatality more than mean: ',  sep = '\n')
fatality_more_than_mean.head(10)

The states with fatality more than mean: 


S. No.           Name of State / UT  Total Confirmed cases  Active  \
1        2  Andaman and Nicobar Islands                     33       1   
2        3            Arunachal Pradesh                      1       0   
4        5                        Bihar                    517     396   
5        6                   Chandigarh                     97      77   
6        7                 Chhattisgarh                     57      21   
7        8                        Delhi                   4549    3123   
8        9                          Goa                      7       0   
10      11                      Haryana                    442     192   
12      13            Jammu and Kashmir                    701     406   
15      16                       Kerala                    500      95   

    Cured/Discharged/Migrated  Deaths  Fatality rate  
1                          32       0       0.000000  
2                           1       0       0.000000  
4                         117       4       0.007737  
5                          19       1       0.010309  
6                          36       0       0.000000  
7                        1362      64       0.014069  
8                           7       0       0.000000  
10                        245       5       0.011312  
12                        287       8       0.011412  
15                        401       4       0.008000

# Determining the mortality ratio

In [23]:
#death_rows = selecting_rows[:,-1:]
#death_rows = death_rows.flatten()

In [24]:
mortality_ratio = 42546 / 11775
print('The mortality rate is: ', mortality_ratio, sep = '\n')

The mortality rate is: 
3.613248407643312


In [25]:
total_case = selecting_rows[:,0:-3]
cured_rows = selecting_rows[:,2:-1]
mortality_ratio_state =  cured_rows / total_case 

In [26]:
mortality_1 = np.hstack([selecting_rows, mortality_ratio_state])

print('The new array with volume column is: ', mortality_1[:4], sep = '\n')

The new array with volume column is: 
[[1.58300000e+03 1.06200000e+03 4.88000000e+02 3.30000000e+01
  3.08275426e-01]
 [3.30000000e+01 1.00000000e+00 3.20000000e+01 0.00000000e+00
  9.69696970e-01]
 [1.00000000e+00 0.00000000e+00 1.00000000e+00 0.00000000e+00
  1.00000000e+00]
 [4.30000000e+01 9.00000000e+00 3.30000000e+01 1.00000000e+00
  7.67441860e-01]]


In [27]:
file['Mortality Rate'] = mortality_ratio_state
file.head()

S. No.           Name of State / UT  Total Confirmed cases  Active  \
0       1               Andhra Pradesh                   1583    1062   
1       2  Andaman and Nicobar Islands                     33       1   
2       3            Arunachal Pradesh                      1       0   
3       4                        Assam                     43       9   
4       5                        Bihar                    517     396   

   Cured/Discharged/Migrated  Deaths  Fatality rate  Mortality Rate  
0                        488      33       0.020846        0.308275  
1                         32       0       0.000000        0.969697  
2                          1       0       0.000000        1.000000  
3                         33       1       0.023256        0.767442  
4                        117       4       0.007737        0.226306

In [28]:
mortality_mean = np.mean(file, axis = 0)
print('The mean of fatality rate is: ', mortality_mean, sep = '\n')

The mean of fatality rate is: 
S. No.                         16.500000
Total Confirmed cases        1329.562500
Active                        918.000000
Cured/Discharged/Migrated     367.968750
Deaths                         43.468750
Fatality rate                   0.019660
Mortality Rate                  0.475743
dtype: float64


From above we can see that Mortality Rate is greater than Fatality rate.

# Determing the states having more mortality ratio than average

In [29]:
mortality_more_than_mean = file.loc[file["Mortality Rate"] > 0.01966]
print('The states with mortality more than mean: ',  sep = '\n')
mortality_more_than_mean.head(10)

The states with mortality more than mean: 


S. No.           Name of State / UT  Total Confirmed cases  Active  \
0       1               Andhra Pradesh                   1583    1062   
1       2  Andaman and Nicobar Islands                     33       1   
2       3            Arunachal Pradesh                      1       0   
3       4                        Assam                     43       9   
4       5                        Bihar                    517     396   
5       6                   Chandigarh                     97      77   
6       7                 Chhattisgarh                     57      21   
7       8                        Delhi                   4549    3123   
8       9                          Goa                      7       0   
9      10                      Gujarat                   5428    4096   

   Cured/Discharged/Migrated  Deaths  Fatality rate  Mortality Rate  
0                        488      33       0.020846        0.308275  
1                         32       0       0.000000        0.969697  
2                          1       0       0.000000        1.000000  
3                         33       1       0.023256        0.767442  
4                        117       4       0.007737        0.226306  
5                         19       1       0.010309        0.195876  
6                         36       0       0.000000        0.631579  
7                       1362      64       0.014069        0.299406  
8                          7       0       0.000000        1.000000  
9                       1042     290       0.053427        0.191968